Add CI workflow for deploying typedocs

CDThomas opened the pull request:
This PR adds a workflow to CI for deploying the generated typedocs to Vercel. This workflow is based on the existing workflow for deploying docs.

Note that I've removed `stashjs-grpc` from the generated docs. This is because we're moving this package back into the private repo (to make life easier for automating releases of the generated JS gRPC client code).

Also note that the typedocs generated from this repo are deployed to a new Vercel project. The docs generated from this repo won't show up on [docs.cipherstash.com/tsdoc](https://docs.cipherstash.com/tsdoc) until we've added the path rewrite to the Vercel project for [docs.cipherstash.com](docs.cipherstash.com) in the private repo.

bennetthardwick reviewed (approved):
Lgtm, that baseurl thing might give you some issues with the Vercel proxy - might have to just put the tsdoc on a subdomain instead.

CDThomas commented:
> Lgtm, that baseurl thing might give you some issues with the Vercel proxy - might have to just put the tsdoc on a subdomain instead.

Ah yeah, good point @bennetthardwick. I suppose that we could also add `/tsdoc` to the path in this project the same way that we did before. A bit gross but that could be an option if we want to keep the paths the same. A separate subdomain would be nice and simple though.

Any preferences between docs.ciphserstash.com/tsdoc and something like tsdoc.cipherstash.com @auxesis @freshtonic?

auxesis commented:
> Any preference between docs.ciphserstash.com/tsdoc and something like tsdoc.cipherstash.com @auxesis @freshtonic?

Let's please keep it to https://docs.cipherstash.com/tsdoc (where it's currently deployed).

You can see how the documentation generation and deployment is set up right now [over here](https://github.com/cipherstash/platform/blob/b13863b9ff5243bbb0c4f8986b72eb71b01b3a8c/.github/workflows/deploy_docs.yml#L36-L49).

CDThomas commented:
I've updated the typedocs to live under the `/tsdoc` path (similar to what the existing setup does).

I've also added in the docs for `stash-rs` and `stashjs-typeorm`.

I think this is good to go, but I'll test these changes out against the (upcoming) Platform changes before merging.

bennetthardwick commented on build.sh:
```diff
@@ -21,7 +21,9 @@ subproject_setup() {
 subproject_build() {
   pnpm install --frozen-lockfile
-  pnpm -r build
+  pnpm --recursive --filter "!./packages/stash-typedoc" build
+  # Build stash-typedoc last because it depends on the other packages already being built
+  pnpm --filter "./packages/stash-typedoc" build
```
If you made all the packages dev dependencies it would evaluate in the correct order. Not a big deal though.

CDThomas replied:
Nice, that sounds like a better solution. I'll fix that up.

bennetthardwick reviewed (approved):
Awesome 🔥

auxesis reviewed (approved):
LGTM. 

Nice work @CDThomas.
